internal/app/ui/cpwsarea/wsmap/pmap/psettings/screenshot.go

```diff
@@ -57,12 +63,30 @@ func (p *Panel) showScreenshot() {
 
 func (p *Panel) createScreenshot() {
 	p.sessionScreenshot.saving = true
-
-	width, height := p.editor.Dmm().MaxX*dmmap.WorldIconSize, p.editor.Dmm().MaxY*dmmap.WorldIconSize
+	selectedTool := tools.Selected()
+	grabCurrentlySelected := selectedTool.Name() == tools.TNGrab
+	boundX, boundY := float32(0), float32(0)
+	var width, height int
+	if p.sessionScreenshot.inSelectionMode && grabCurrentlySelected && selectedTool.(*tools.ToolGrab).HasSelectedArea() {
+		bounds := selectedTool.(*tools.ToolGrab).Bounds() //get grab tool bounds so we can calculate boundX and boundY
+		width, height = (int(bounds.X2-bounds.X1)+1)*dmmap.WorldIconSize, (int(bounds.Y2-bounds.Y1)+1)*dmmap.WorldIconSize
+		boundX = -float32((int(bounds.X1) - 1) * dmmap.WorldIconSize) //now change bounds so we can use them in Translate
+		boundY = -float32((int(bounds.Y1) - 1) * dmmap.WorldIconSize)
+	} else if p.sessionScreenshot.inSelectionMode && !grabCurrentlySelected || !selectedTool.(*tools.ToolGrab).HasSelectedArea() {
```

Comment:
Checks are a little bit over complicated. Editor crashed after I've tried to do a full screenshot: 

```
panic: interface conversion: tools.Tool is *tools.ToolAdd, not *tools.ToolGrab

goroutine 1 [running, locked to thread]:
sdmm/internal/app/ui/cpwsarea/wsmap/pmap/psettings.(*Panel).createScreenshot(0xc000d6bfb0)
	sdmm/internal/app/ui/cpwsarea/wsmap/pmap/psettings/screenshot.go:75 +0x4a6
sdmm/internal/imguiext/widget.(*ButtonWidget).Build(0xc004c88a20)
	sdmm/internal/imguiext/widget/button.go:158 +0x51f
sdmm/internal/imguiext/widget.Layout.BuildV({0xc004c7b010?, 0x1, 0xc00011b730?}, 0xc00011b730?)
	sdmm/internal/imguiext/widget/layout.go:58 +0x111
sdmm/internal/imguiext/widget.Layout.Build({0xc004c7b010?, 0xc00011b7a0?, 0x10000f52a?})
	sdmm/internal/imguiext/widget/layout.go:11 +0x3f
sdmm/internal/imguiext/widget.(*disabledWidget).Build(0xc002578ec0)
	sdmm/internal/imguiext/widget/disabled.go:12 +0x51
sdmm/internal/imguiext/widget.Layout.BuildV({0xc00011b840?, 0x1, 0x1005a6d20?}, 0xc004c7af01?)
	sdmm/internal/imguiext/widget/layout.go:58 +0x111
sdmm/internal/imguiext/widget.Layout.Build({0xc00011b840?, 0x10068c390?, 0x1009c4820?})
	sdmm/internal/imguiext/widget/layout.go:11 +0x3f
sdmm/internal/app/ui/cpwsarea/wsmap/pmap/psettings.(*Panel).showScreenshot(0xc000d6bfb0)
	sdmm/internal/app/ui/cpwsarea/wsmap/pmap/psettings/screenshot.go:60 +0x4be
sdmm/internal/app/ui/cpwsarea/wsmap/pmap/psettings.(*Panel).Process(0xc00011b8d8?)
	sdmm/internal/app/ui/cpwsarea/wsmap/pmap/psettings/psettings.go:46 +0x45
sdmm/internal/app/ui/cpwsarea/wsmap/pmap.(*PaneMap).showPanelV(0xc000d229c0, {0xc00011b9b0, 0x17}, 0x1, 0xa0?, 0xc00011ba20)
	sdmm/internal/app/ui/cpwsarea/wsmap/pmap/panel.go:64 +0x2b2
sdmm/internal/app/ui/cpwsarea/wsmap/pmap.(*PaneMap).Process(0xc000d229c0)
	sdmm/internal/app/ui/cpwsarea/wsmap/pmap/pmap.go:223 +0x2f9
sdmm/internal/app/ui/cpwsarea/wsmap.(*WsMap).Process(0xc000000001?)
	sdmm/internal/app/ui/cpwsarea/wsmap/wsmap.go:77 +0x1d
sdmm/internal/app/ui/cpwsarea/workspace.(*Workspace).Process(...)
	sdmm/internal/app/ui/cpwsarea/workspace/workspace.go:82
sdmm/internal/app/ui/cpwsarea.(*WsArea).showWorkspaceWindow(0xc0000d2580, 0xffffffff8b93e3bd, 0xc0010679e0)
	sdmm/internal/app/ui/cpwsarea/process.go:65 +0x1af
sdmm/internal/app/ui/cpwsarea.(*WsArea).processWorkspaces(0xc0000d2580, 0xc0000d2500?)
	sdmm/internal/app/ui/cpwsarea/process.go:29 +0xec
sdmm/internal/app/ui/cpwsarea.(*WsArea).Process(0xc0000d2580, 0x5ed6fc?)
	sdmm/internal/app/ui/cpwsarea/process.go:17 +0x56
sdmm/internal/app/ui/layout.(*Layout).wrapNodeV(0xc0000d2500?, {0x1005efe24?, 0x9?}, 0x689b40?, {0x10068d870?, 0xc0000d2580?}, {0x8?, 0x3f?})
	sdmm/internal/app/ui/layout/layout.go:188 +0x3a
sdmm/internal/app/ui/layout.(*Layout).showWorkspaceAreaNode(...)
	sdmm/internal/app/ui/layout/layout.go:125
sdmm/internal/app/ui/layout.(*Layout).Process(0xc0000d2500)
	sdmm/internal/app/ui/layout/layout.go:91 +0xff
sdmm/internal/app.(*app).Process(0xc000132a00)
	sdmm/internal/app/app.go:140 +0x4a
sdmm/internal/app/window.(*Window).runFrame(0xc00010fb90)
	sdmm/internal/app/window/process.go:33 +0x31
sdmm/internal/app/window.(*Window).Process(0xc00010fb90)
	sdmm/internal/app/window/process.go:26 +0x29
sdmm/internal/app.Start()
	sdmm/internal/app/app.go:65 +0x7f8
main.main()
	sdmm/main.go:10 +0x19
```

I did nothing, but opened a map and clicked screenshot button. Default tool, default configurations.

From what I see you at least need to do next:
```go
if p.sessionScreenshot.inSelectionMode {
    if !grabCurrentlySelected || !selectedTool.(*tools.ToolGrab).HasSelectedArea() {
        // error dialog
        return
    } else {
        // logic
    }
} else {
    width, height = p.editor.Dmm().MaxX*dmmap.WorldIconSize, p.editor.Dmm().MaxY*dmmap.WorldIconSize
}
```

Reply:
done